src/LocationTracker.ts

```diff
@@ -0,0 +1,93 @@
+import Locations, { LocationChange } from './Locations.js';
+import { SpaceMember } from './Space.js';
+import { EventListener } from './utilities/EventEmitter.js';
+
+/**
+ * Responds to a locationChange with:
+ * - `true` if the change should trigger an event to fire
+ * - `false` if the change should not trigger an event to fire
+ *
+ * @param {LocationChange<T>} locationChange
+ * @return {boolean}
+ */
+export type LocationTrackerPredicate<T> = (locationChange: LocationChange<T>) => boolean;
+
+/** Class that enables tracking and responding to changes on specific location changes. */
+export default class LocationTracker<T> {
+  private locationListenerWrappers: Map<EventListener<any>, EventListener<any>>;
```

Comment:
Should this not be `Map<EventListener<T>, EventListener<T>>`?

Reply:
Yes, I've changed to `Map<EventListener<T>, EventListener<LocationChange<T>>` because the listenerWrapper takes a LocationChange<T> object from the event